src/WebJobs.Extensions.DurableTask/HttpApiHandler.cs

```diff
@@ -212,10 +214,14 @@ public async Task<HttpResponseMessage> HandleRequestAsync(HttpRequestMessage req
                         string entityName = (string)routeValues[EntityNameRouteParameter];
                         string entityKey = (string)routeValues[EntityKeyRouteParameter];
                         EntityId entityId = new EntityId(entityName, entityKey);
-                        if (request.Method == HttpMethod.Get)
+                        if (!string.IsNullOrEmpty(entityKey) && request.Method == HttpMethod.Get)

```

Comment:
Unfortunately this means it won't be possible to query a single entity with an empty `entityKey` value (which is something we intentionally support). I'm not sure the best way to fix this.  Maybe if we get a request for `GET /entities/$` we could treat the `$` internally as the empty string? I don't think we need to block on this for now, but something to consider. If we don't fix it, we should add a TODO comment explaining that we need to revisit this.

Reply:
My understanding was that you required an entityKey for that, that's why you pass in an EntityId, which throws an exception when you try to initialize one without an entityKey.

Reply:
It throws for `null` but not for the empty string.